geoips/plugins/modules/readers/abi_netcdf.py

```diff
@@ -509,6 +510,94 @@ def call(fnames, metadata_only=False, chans=None, area_def=None, self_register=F
     """
     Read ABI NetCDF data from a list of filenames.
 
+    Parameters
+    ----------
+    fnames : list
+        * List of strings, full paths to files
+    metadata_only : bool, default=False
+        * Return before actually reading data if True
+    chans : list of str, default=None
+        * List of desired channels (skip unneeded variables as needed).
+        * Include all channels if None.
+    area_def : pyresample.AreaDefinition, default=None
+        * Specify region to read
+        * Read all data if None.
+    self_register : str or bool, default=False
+        * register all data to the specified dataset id (as specified in the
+          return dictionary keys).
+        * Read multiple resolutions of data if False.
+
+    Returns
+    -------
+    dict of xarray.Datasets
+        * dictionary of xarray.Dataset objects with required Variables and
+          Attributes.
+        * Dictionary keys can be any descriptive dataset ids.
+
+    See Also
+    --------
+    :ref:`xarray_standards`
+        Additional information regarding required attributes and variables
+        for GeoIPS-formatted xarray Datasets.
+    """
+    all_metadata = [
+        call_single_time([x], metadata_only=True)["METADATA"] for x in fnames
+    ]
+    if metadata_only:
+        return readers.concatenate_metadata(all_metadata)
```

Comment:
It looks like you use "concatenate_metadata" for metadata_only, but not for the actual data reads. I think we want to ensure all the source_file_attributes, source_file_names, and source_file_datetimes metadata gets added to the xarray datasets for a full file read as well as "metadata_only"

I think we want to store source_file_names, source_file_attributes, and source_file_datetimes.

Would something like this work:

all_metadata = readers.concatenate_metadata([
        call_single_time([x], metadata_only=True)["METADATA"] for x in fnames
    ])

Just concatenate the metadata before the metadata_only if statement, not within it?

Reply:
Yup that would work! I'll make that change now. Good point.

Reply:
Done!

Reply:
Thanks!